src/prompt-input/internal.tsx

```diff
@@ -0,0 +1,168 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { Ref, useCallback, useEffect, useRef } from 'react';
+import { getBaseProps } from '../internal/base-component';
+import InternalButton from '../button/internal';
+import { fireKeyboardEvent, fireNonCancelableEvent } from '../internal/events';
+import { PromptInputProps } from './interfaces';
+import { useFormFieldContext } from '../internal/context/form-field-context';
+import useForwardFocus from '../internal/hooks/forward-focus';
+import clsx from 'clsx';
+import styles from './styles.css.js';
+import { InternalBaseComponentProps } from '../internal/hooks/use-base-component';
+import { convertAutoComplete } from '../input/utils';
+import { useDensityMode } from '@cloudscape-design/component-toolkit/internal';
+
+export interface InternalPromptInputProps extends PromptInputProps, InternalBaseComponentProps {}
+
+const InternalPromptInput = React.forwardRef(
+  (
+    {
+      value,
+      actionButtonAriaLabel,
+      actionButtonIconName,
+      actionButtonIconUrl,
+      actionButtonIconSvg,
+      actionButtonIconAlt,
+      ariaLabel,
+      autoComplete = true,
+      autoFocus,
+      disableActionButton,
+      disableBrowserAutocorrect,
+      disabled,
+      maxRows,
+      minRows,
+      name,
+      onAction,
+      onBlur,
+      onChange,
+      onFocus,
+      onKeyDown,
+      onKeyUp,
+      placeholder,
+      readOnly,
+      spellcheck,
+      __internalRootRef = null,
+      ...rest
+    }: InternalPromptInputProps,
+    ref: Ref<PromptInputProps.Ref>
+  ) => {
+    const { ariaLabelledby, ariaDescribedby, controlId, invalid, warning } = useFormFieldContext(rest);
+
+    const baseProps = getBaseProps(rest);
+
+    const textareaRef = useRef<HTMLTextAreaElement>(null);
+
+    const isCompactMode = useDensityMode(textareaRef) === 'compact';
+
+    useForwardFocus(ref, textareaRef);
+
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLTextAreaElement>) => {
+      if (onKeyDown) {
+        fireKeyboardEvent(onKeyDown, event);
+      }
+
+      if (event.key === 'Enter' && !event.shiftKey) {
+        if ('form' in event.target && event.target.form !== null && !event.isDefaultPrevented()) {
+          (event.target.form as HTMLFormElement).requestSubmit();
+        }
+
+        event.preventDefault();
+        fireNonCancelableEvent(onAction, { value });
+      }
+    };
+
+    const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+      fireNonCancelableEvent(onChange, { value: event.target.value });
+      adjustTextareaHeight();
+    };
+
+    const adjustTextareaHeight = useCallback(() => {
+      const PADDING = 4;
```

Comment:
it should be possible to import design tokens as JS values

https://github.com/cloudscape-design/components/blob/main/src/app-layout/visual-refresh/main.tsx#L10

Reply:
I saw that, but noticed that the only tokens in that file are for charts and a few for spacing. I did a surface-level search and found a couple other places where we declare padding as a number value:
- https://github.com/cloudscape-design/components/blob/main/src/code-editor/pane.tsx#L17
- https://github.com/cloudscape-design/components/blob/main/src/mixed-line-bar-chart/bar-series.tsx#L74

But I agree that this seems like it should pull from the token value. Perhaps a larger team thread should be had?

Reply:
I've updated it and I agree I like it better when both values pull from the same source.